diff --git a/trim.module b/trim.module
index 5217063..eeb3382 100644
--- a/trim.module
+++ b/trim.module
@@ -11,6 +11,15 @@ use Drupal\Core\Form\FormStateInterface;
  * Implements hook_form_alter().
  */
 function trim_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Do trimming only for content entity types.
+  $form_object = $form_state->getformObject();
+  if (!method_exists($form_object, 'getEntity')) {
+    return;
+  }
+  $entity_type_class = get_class($form_object->getEntity()->getEntityType());
+  if ($entity_type_class != 'Drupal\Core\Entity\ContentEntityType') {
+    return;
+  }
   // Ensure that there is an array here.
   if (!isset($form['#validate'])) {
     $form['#validate'] = [];
