diff --git a/modules/media_library/src/MediaLibraryEditorOpener.php b/modules/media_library/src/MediaLibraryEditorOpener.php
index 0c0af98..3cf838b 100644
--- a/modules/media_library/src/MediaLibraryEditorOpener.php
+++ b/modules/media_library/src/MediaLibraryEditorOpener.php
@@ -70,7 +70,6 @@ class MediaLibraryEditorOpener implements MediaLibraryOpenerInterface {
       'attributes' => [
         'data-entity-type' => 'media',
         'data-entity-uuid' => $selected_media->uuid(),
-        'data-align' => 'center',
       ],
     ];
     $response->addCommand(new EditorDialogSave($values));
