diff --git a/sftouchscreen.js b/sftouchscreen.js
index 8364cf4..9af28cf 100644
--- a/sftouchscreen.js
+++ b/sftouchscreen.js
@@ -20,6 +20,8 @@
       breakpointUnit: 'px',
       useragent: '',
       behaviour: 2,
+      clonePrefix: '',
+      cloneSuffix: '',
       disableHover: false
     }, options);
 
@@ -39,6 +41,7 @@
             cloneLink = parent.children('a.menuparent,span.nolink.menuparent').clone(),
             // Wrapping the hyerplinks in <li>.
             cloneLink = $('<li class="sf-clone-parent" />').html(cloneLink);
+            cloneLink.find('a').prepend(options.clonePrefix).append(options.cloneSuffix);
             // Removing unnecessary stuff.
             cloneLink.find('.sf-sub-indicator').remove(),
             // Adding a helper class and attaching them to the sub-menus.
